packages/schematics/angular/web-worker/index.ts

```diff
@@ -140,7 +140,8 @@ function addSnippet(options: WebWorkerOptions): Rule {
 
     const siblingModules = host.getDir(options.path).subfiles
       // Find all files that start with the same name, are ts files, and aren't spec files.
-      .filter(f => f.startsWith(options.name) && f.endsWith('.ts') && !f.endsWith('spec.ts'))
+      .filter(f => f.startsWith(options.name)
```

Comment:
@filipesilva should web-workers should be inserted only to components?

@santoshyadav198613 - the statements here can be simplified slightly, if the above is the case.
```ts
.filter(f => f.startsWith(options.name) && f.endsWith('component.ts'))
```

Reply:
Now that you mention it, it's very common to insert them into services as well I believe. I think we should specialcase spec files and routing files actually...

So something like this:
```
.filter(f => f.startsWith(options.name) && f.endsWith('.ts') 
  && !f.endsWith('spec.ts') && !f.endsWith('-routing.module.ts'))
```

These two are indeed special because our generators make them automatically.

Reply:
But in that case, we have interfaces, directives as well.

Reply:
Yes, it should be ok to put a webworker in those. Specs and routing files are different because are auxiliary to the other, more important, file. And in this specific code that is problematic because we sort by filename.

Reply:
thanks a lot, will make the changes. You guys are super helpful.

Reply:
Also i think we should not consider moduel at all so changing it to
 .filter(f => f.startsWith(options.name) && f.endsWith('.ts') 
  && !f.endsWith('spec.ts') && !f.endsWith('-.module.ts')) 
would be better.

Reply:
I think that's a fair assumption, yes. That looks good to me.

Reply:
Done with the changes.